department-of-veterans-affairs/caseflow
# Remove eligible_due_to_date for aod motions

Resolves #14085

### Description
Remove old check that allowed a motion to be associated with an appeal because the motion was created after the appeal's receipt date. We have now [associated every motion with an appeal](https://github.com/department-of-veterans-affairs/caseflow/issues/14085#issuecomment-692347472) and this incorrect check should be removed.

### Acceptance Criteria
Appeals are only considered aod due to motion if there is an 
- [ ] age related motion that applies to any appeal with the same claimant
- [ ] any other reason related motion that applies to that exact appeal

### Testing Plan
1. Sign in as AOD_USER and find any non aod ama case
```ruby
appeal = Appeal.find_by(uuid: "173b6ce2-070c-4eec-a463-301f8f18c70e")
appeal.aod?
=> false
```
2. Find the "Edit" link next to the appeal stream type
1. Grant an AOD motion for any reason other than age
```ruby
appeal.aod?
=> true
```
4. Remove the appeal from the aod motion
```ruby
motion = AdvanceOnDocketMotion.find_by(appeal: appeal)
motion.update_attribute('appeal_id', nil)
AdvanceOnDocketMotion.find_by(appeal: appeal)
=> nil
appeal.aod?
=> false
AdvanceOnDocketMotion.first.created_at > appeal.receipt_date
=> true
```
5. Ensure aod does not show up in the ui by reloading case details
